packages/http-client-java/generator/http-client-generator/src/main/java/com/microsoft/typespec/http/client/generator/mapper/TypeSpecClientMapper.java

```diff
@@ -63,4 +65,21 @@ protected List<String> getModelsPackages(List<ClientModel> clientModels, List<En
 
         return new ArrayList<>(packages);
     }
+
+    @Override
+    protected String getResponseHeaderName(Header header) {
+        String clientName;
+        if (header.getLanguage() != null
+            && header.getLanguage().getJava() != null
+            && !CoreUtils.isNullOrEmpty(header.getLanguage().getJava().getName())) {
```

Comment:
Where do we set header.language().java()?

Reply:
We haven't.

Just keep this logic, in case we later actually set it.